### PR TITLE
bug(template): more informative error when no template defined in pipeline

### PR DESCRIPTION
**Pull request by KellyMerrick**
For https://github.com/go-vela/community/issues/327.

When a template isn't defined in a pipeline as in this example:
```
version: "1"

steps:
  - name: some-step
    template:
      name: this_template_isnt_defined
      vars:
        some: var
```

A user will now receive a more useful error in the Audit tab, where `some-step` is the step name in a pipeline:
```
missing template source for this_template_isnt_defined in pipeline for step some-step
```

image

Previously, the user was receiving an ambiguous error such as:
image

This was implemented as such:
https://github.com/go-vela/compiler/blob/73f56ab4833008721dc0067d6a5e7b9730a821e8/compiler/native/expand.go#L53-L64

### CODE REVIEW

File: compiler/native/expand.go [hunk 1]
```diff
@@ -50,14 +50,19 @@ func (c *client) ExpandSteps(s *yaml.Build, tmpls map[string]*yaml.Template) (ya
...
+		if len(step.Template.Name) == 0 {
 			// add existing step if no template
 			steps = append(steps, step)
 			continue
 		}
 
+		// lookup step template name
+		tmpl, ok := tmpls[step.Template.Name]
+		if !ok {
+			return yaml.StepSlice{}, yaml.SecretSlice{}, yaml.ServiceSlice{}, fmt.Errorf("missing template source in pipeline for step %s", step.Name)
```
Comment (wass3r) [inline compiler/native/expand.go:63]: could we expose the template name that was used in the error message? would that be helpful?

**Review by wass3r — COMMENTED** [review]
looks good.. just a comment for potential enhancement. let me know what you think.

File: compiler/native/expand.go (hunk 1 above)
Comment (KellyMerrick) [reply compiler/native/expand.go]: Good idea. Updated.

**Review by wass3r — APPROVED** [review]

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**